Add OTEL_RESOURCE_ATTRIBUTES paragraph to OTel tracing docs (#4718)
langchain-ai/docs: src/langsmith/trace-with-opentelemetry.mdx

OTEL_EXPORTER_OTLP_ENDPOINT: Override the endpoint URL
OTEL_EXPORTER_OTLP_HEADERS: Add custom headers (LangSmith API keys and Project are added automatically)
OTEL_SERVICE_NAME: Set a custom service name (defaults to "langsmith")
OTEL_RESOURCE_ATTRIBUTES: Attach custom metadata fields at the process level (see below)
```

### Add process-level resource attributes

> You can attach custom metadata to every trace emitted by a process using the standard OpenTelemetry `OTEL_RESOURCE_ATTRIBUTES` environment variable. Unlike span-level attributes (set per-span in code), resource attributes are set once at the process level and automatically propagated to all spans — making them ideal for any custom metadata you want to tag your traces with, such as a username, request ID, environment, or deployment version.

Reviewer: ```suggestion
You can attach custom metadata to every trace emitted by a process using the standard OpenTelemetry `OTEL_RESOURCE_ATTRIBUTES` environment variable. Unlike span-level attributes (set per-span in code), resource attributes are set once at the process level and automatically propagated to all spans. This makes them ideal for any custom metadata you want to tag your traces with, such as a username, request ID, environment, or deployment version.
```

The value is a comma-separated list of `key=value` pairs:

```bash
OTEL_RESOURCE_ATTRIBUTES="username=abc,id=1,environment=production"
```

You can use any custom keys you like, or follow the [OpenTelemetry resource semantic conventions](https://opentelemetry.io/docs/specs/semconv/resource/) for standard fields such as `deployment.environment`, `service.version`, and `cloud.region`.

> These attributes appear in LangSmith under the trace's metadata and can be used to filter and group traces across your workspace.

Reviewer 2: Should we specify that it's under `otel.resource.` prefix?

> These attributes appear in LangSmith under the trace's metadata and can be used to filter and group traces across your workspace.

Reviewer: ```suggestion
These attributes appear in LangSmith under the trace's metadata, which you can use to filter and group traces across your workspace.
```
